core/tx_pool.go

```diff
@@ -622,14 +626,14 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 		return ErrInvalidSender
 	}
 
-	// Ensure the fee currency is native or whitelisted.
-	if tx.FeeCurrency() != nil && !currency.IsWhitelisted(*tx.FeeCurrency(), nil, nil) {
+	gasPriceMinimum, isWhitelisted := pool.currentCtx.GetGasPriceMinimum(tx.FeeCurrency())
```

Comment:
It would be nice to differentiate between failures due to a currency not being whitelisted and the GPM for that currency not being available. I would recommend we add the `IsWhitelisted` function to block context to use here.

Reply:
removed gpm check; kept only isWhitelisted